tests/dragonfly/replication_test.py

```diff
@@ -2033,13 +2033,19 @@ async def test_client_pause_with_replica(df_factory, df_seeder_factory):
     assert await seeder.compare(capture, port=replica.port)
 
 
+@pytest.mark.debug_only
+@dfly_args({"proactor_threads": 2})
```

Comment:
Fixes #5662


1. This test takes significant amount of time in `debug mode` 100+ seconds
2. This test is extremely inefficient when it runs on release mode because loading and saving a snapshot is orders of magnitudes faster in release than in debug. So when we try to assert that that dragonfly is in loading state we need to have a big snapshot for release. Guess what! This kills the running time in debug mode.

The fix:

Only run this test in debug mode where generally loading a snpashot is `slower` as we only care for behavioural semantics here. That way the test has more time to check the `LOADING` state without adjusting the seeder/snapshot size on release builds. 

Bonus points is now we can reduce the snapshot size and make the debug test run faster!